datumaro/components/operations.py

```diff
@@ -132,36 +69,65 @@ class ExactMerge:
         - in case of conflicts, throws an error
     """
 
-    @classmethod
-    def merge(cls, *sources: IDataset) -> DatasetItemStorage:
+    def __init__(self, **options):
+        super().__init__(**options)
+
+    def merge(self, *sources: IDataset) -> DatasetItemStorage:
         items = DatasetItemStorage()
         for source_idx, source in enumerate(sources):
             for item in source:
                 existing_item = items.get(item.id, item.subset)
                 if existing_item is not None:
                     try:
-                        item = cls._merge_items(existing_item, item)
+                        item = self._merge_items(existing_item, item)
                     except DatasetMergeError as e:
                         e.sources = set(range(source_idx))
                         raise e
 
                 items.put(item)
         return items
 
-    @classmethod
-    def _merge_items(cls, existing_item: DatasetItem, current_item: DatasetItem) -> DatasetItem:
+    def merge_infos(self, sources: Iterable[IDataset]) -> DatasetInfo:
+        return self._merge_infos(sources)
+
+    def merge_categories(self, sources: Iterable[IDataset]) -> CategoriesInfo:
```

Comment:
I think that `sources: Iterable[IDataset]` -> `sources: Iterable[CategoriesInfo]` as well as elsewhere, e.g. `IntersectMerge`, ....